ITREX-602

GarondEisenfaust opened the pull request:
Changed videopool endpoint from /videos to videos/course
Changed download endpoint form /videos to videos/download
Added an endpoint that receives a JSON list of UUIDs and returns a list of Video entities that have those ids as primary keys 

I had to change those endpoints because they led to conflicts with the new one. This is a breaking change.

GarondEisenfaust committed:
Changed videopool endpoint to videos/course
Changed download endpoint to videos/download
Added endpoint to get a list of Videos by a list of ids

Well5a commented:
gibt es auch ein entsprechendes Ticket im Frontend, das das anpasst?

dgwetzel commented:
Habe ein Issue erstellt.

https://it-rex.atlassian.net/browse/ITREX-603

swjat commented on src/main/java/de/uni_stuttgart/it_rex/media/web/rest/written/VideoResource.java:
```diff
@@ -244,10 +246,25 @@ private ByteRangeDTO parseByteRange(final HttpHeaders headers,
+   */
+  @GetMapping("/videos")
+  public Map<UUID, Video> findAllWithIds(
+      @RequestBody final List<UUID> videoIds) {
```
GET-requests mit body sind eig nicht erlaubt.

GarondEisenfaust replied:
Dann überleg ich mir was anderes

GarondEisenfaust committed:
Changed endpoint to get video pool to /videos/course/{courseId:.+}
Changed endpoint to get a map of videos to use url params instead of the body to transfer the ids

swjat reviewed (approved):
bueno
